lib/pluginmanager/sources/http.rb

```diff
@@ -0,0 +1,78 @@
+# encoding: utf-8
+require "rubygems/package"
+require "pluginmanager/sources/base"
+require "net/http"
+
+module LogStash::PluginManager::Sources
+
+  class HTTP < Base
+
+    attr_reader :uri, :fallback, :header_options
+
+    ##
+    # Fallback URL used by default when installing packs given
+    # by name, if you require to install a pack from a custom URL
+    # you will need to pass the full URL.
+    ##
+    ROOT_URI = "https://download.elastic.co".freeze
+
+    ##
+    # User agent string used to identify the requester is a
+    # LogStash agent
+    ##
+    USER_AGENT = "Logstash/#{LOGSTASH_VERSION}".freeze
+
+    def initialize(uri)
+      super
+      @fallback = !uri.start_with?("http")
+      if fallback
+        @uri = URI(fallback_uri(uri))
+      else
+        @uri = URI(uri)
+      end
```

Comment:
OK, this part of the initialize was confusing me, the parent class set the `@uri` properties when we pass the `value` to the method call. So if this is the case we could rewrite this method like this.

```ruby
def initialize(uri)
  @fallback = !uri.start_with?("http")

  if fallback
    @uri = super(fallback_uri(uri))
  else
    @uri = super(@uri)
  end
  @header_options = {'User-Agent' => USER_AGENT}
end
```